Enable the VM to be used on Amazon AWS

theabro opened the pull request:
It is fairly straight-forward to use Vagrant with Amazon AWS to allow the VM to work on there. An Ubuntu 18.04 VM has been tested and works in a comparable way to VirtualBox. Requirements and usage instructions have been added to the README file.

dpmatthews committed:
wsBcBAABCAAQBQJaJ+6xCRBK7hj4Ov3rIwAAdHIIAIlqTsk9YKreRubw3q18F//m
 z7nWV51Z/klSLhe2Cx63xMkhq1MH/DDTKvV7aigzUZuMQH3DvBbQ3vyRjWvDg9km
 64+3tOAosaD3bmfBP2l11qiH1/ccpVnDdxvkVGg62rViX6F9CM021OP/FP/uJm8r
 4moevaGwTzXmFZY3QRZfteBUqkcw9B7YTFK2Rfe6qobgTruQxCWOvKbLAeY3azEt
 f4bE1Sxf1D0/50EnGFjFHZ27DK1AkPzOWs0ihUNiKQaaM6VcuSp++6svZXE7UtZE
 +Kb0VJARQHIrCA7GXW4xBGf5i8EjhUS09wuoxhJnzoUHMV2tGFhHXVCKKYOd74s=
 =IDVE
 -----END PGP SIGNATURE-----
 

Merge pull request #12 from paulcresswell/umdp_support

Add support for building the UMDPs

dpmatthews committed:
Replace Fedora 26 with 27
Replace Ubuntu 1610 with 1710
Centos: use boxes which automatically use the latest minor version
Centos 6/7 + Ubuntu 1404: Use subversion 1.9 rather than 1.8
Ubuntu 1604: Use WANDisco subversion packages (more up to date)
Enable middle mouse button emulation
Use xxdiff rather than tkdiff/kdiff3 where possible
Use leafpad (emacs on Centos) rather than gedit

dpmatthews committed:
Ubuntu 1604: Correct libapache2-svn install error

dpmatthews committed:
Setup local documentation
README.md: Add Troubleshooting section and document vagrant-vbguest plugin
README.md: Fix links to Rose documentation

paulcresswell committed:
Use Cylc 7.6 option to avoid returning control to the script early

dpmatthews committed:
wsBcBAABCAAQBQJalZz6CRBK7hj4Ov3rIwAAdHIIAF0ZvIys7IGTKqF34FBRajUJ
 7b1MHdds7vByFK/1qXwMInJsP6frK95blpKVlWECeX8XKHjH2Zu6RbS/zPw1mTi3
 xKY3JNuKfJnYW5QC9bKWP9+Sp9ppU706hyXL1EblHQJeDjCOSauUgYNMeQdLTe64
 QjrsTnq0D2fTj63Ru7Ano/CGNY4epsug+pUJJuXbp17nSAaoPv1iSZdh6I/Re0Om
 ZJc5GNw/W7zHYh6JQXQ1AZMSxS13cSGLG3VXHfd+f9PIiiTzid17W4wuUVhUiVjE
 6zQHAqYpS6KSswlGYAMX0piFkhQFyo0a9u7dKrjqCJFi50mssO04C1mFbOnuLvY=
 =F1mZ
 -----END PGP SIGNATURE-----
 

Merge pull request #13 from paulcresswell/cylc7.6_fix

Use Cylc 7.6 option to avoid returning control to the um-setup script early

dpmatthews committed:
Replace Fedora 27 with 28 & Ubuntu 1710 with 1804

dpmatthews committed:
Remove some packages which are not required

dpmatthews committed:
Build Rose PDF docs

dpmatthews committed:
Change the default Vagrantfile to load from a named file

matthewrmshin committed:
wsBcBAABCAAQBQJbaWO6CRBK7hj4Ov3rIwAAdHIIAAmPfr0SB79U6MmO9Tl4U5xw
 00OVC1ENtpIn4Z7ufWhdibcdIoBOCpEeLOKyIHNznMkmSQwVXEuIjv/10V+QNqlH
 cm+HuENsXPkeqgAVkYNRtCydHru5LBpoWY4Aa9D7XUEnuAV2pGCXcVGGmhIZx5wt
 UvvUQoRXYgD+USENvwUeyMollZpXTuMkcl2kE8etrs0nlk3Uw46QADlFP60wXRd3
 w60KfRmP+JxhxrNK8a16nbUdsjgacLtwH2oAIzXVDTuRRijr4hIrtHUFxWuwnqOk
 KmHRWudhspkRV4cBwIN5y/K04Vb7nwObqcq4UHdHEkOWcj5crcrVSH1io+v3EbQ=
 =5AyI
 -----END PGP SIGNATURE-----
 

Update README.md

Add DOI badge.

theabro committed:
initial add of Azure Vagrantfile

theabro committed:
attempt 18.04 imagewq

theabro committed:
changed disk

theabro committed:
hard-coded Ubuntu distro

theabro committed:
adding setup info

theabro committed:
problems with conda install of iris (fixed using conda-forge repo) and lxterminal (fixed by switching to xterm)

theabro committed:
need ampersand to return conda command

theabro committed:
changed from original script to take from conda-forge and use xterm

theabro committed:
updated readme

theabro committed:
updated readme

theabro committed:
updated readme

theabro committed:
updated readme

theabro committed:
updated readme

theabro committed:
updated readme

theabro committed:
add to TOC

theabro committed:
updated readme

theabro committed:
updated readme

theabro committed:
updated size

theabro committed:
reset iris install

theabro committed:
reset iris install

theabro committed:
further clarification in README

paulcresswell committed:
Add offline support for shumlib

dpmatthews committed:
Use --purge option with apt-get remove (fix issue with gpg-agent on Ubuntu 1804)
Update minimum versions for VirtualBox & Vagrant
Increase video memory on VirtualBox

dpmatthews committed:
Fix install-iris (change needed to conda install)

paulcresswell committed:
Allow choice of grib_api or eccodes libraries

paulcresswell committed:
Allow a choice of installing EcCodes or GRIB_API

paulcresswell committed:
Add an additional safety check on -v

paulcresswell committed:
Improve help text wording

paulcresswell committed:
Clarify and tidy help text

dpmatthews committed:
wsBcBAABCAAQBQJb4totCRBK7hj4Ov3rIwAAdHIIADOhsvFHzHhqq55VwB0LGr/O
 swloMTrTDXq0EJVIY2Db0CMSAdnLVMKk/5dyDqgk/NuxZYLnfI5D/NKqixSOagAZ
 5qa7pA5ra84rp8naT6HJXejqXhM968orps8xE8wtfMWFGUbwvWE8MffVj1kyJFTC
 zn5bm3hvuy0XUjvBwYCRmc7PSLQdS/7PbaEpS2tIg+nuXQDfGv9LA9DlVOIq8tj8
 VZYVDFxUi6ipWJ9b5CgbY4UZKJrX/ao5Xy1zKw/nUVr1YQDk89VocPXFdKf64WDY
 jlx2OhyWUxG4PGXjuX5f3Z/h9+qJuZgZY5SaJmeRpbZGp/VC0K4UBulWIJj8lWk=
 =ipZe
 -----END PGP SIGNATURE-----
 

Merge pull request #15 from paulcresswell/um11.2-updates

UM11.2 updates

dpmatthews committed:
Change the default VM to Ubuntu 1804

dpmatthews committed:
FCM keywords: add latest keywords

theabro committed:
use 18.04 with a larger Azure VM

theabro committed:
resolved conflict to my changes

paulcresswell committed:
Fix help (wrong name) and clarify no arg. case

paulcresswell committed:
Remove debugging changes

paulcresswell committed:
Fix for --no-detach with latest Rose/Cylc versions

paulcresswell committed:
Update rose bush paths

paulcresswell committed:
Remove path hardwiring

dpmatthews committed:
wsBcBAABCAAQBQJcGgkDCRBK7hj4Ov3rIwAAdHIIAFMqFhFyp0ZHUkDiQQxawM0A
 0h6NyhVQ+JGjK7duTWOIeZekz5nYMeDKLMtj/1X0voquIomlR2V5jnejshL+l9fM
 Ir4c4Xpv55ySmUG0mFCjldPhv+G7I48J0/0ngHWb8TLpmmi0Pc1PpzJ8DcfvSvuG
 nFaJKYzJEKcYw/2S1M5ZyMmL7L4ING5zgvCwK4EZr6pn4STUZ6n7OhUG3qYpU9T1
 DykqBDh3JzZz4sqO6+zEL5tfI4U9BlkX6YQQD9jFvF4YYRMQ+Dnpq1vKYWaKYy9A
 EKkAVsHslkLJTtWD+aM20udfXFwMQjefreCpYFmDXxM2QAW4QudBSNFvEv+nzwo=
 =spm6
 -----END PGP SIGNATURE-----
 

Merge pull request #16 from paulcresswell/um11.2-updates

UM installation fixes

theabro committed:
Merging with latest code in central repo

Merge remote-tracking branch 'upstream/master'

dpmatthews committed:
install-rose-cylc-fcm (was install-latest-versions):
- Update to support cylc Sphinx documentation
- Allow versions to be specified
- Add option to determine whether to set the default version
- Add option to determine whether to build the docs
install-cylc-6: removed
Set desktop background colour

paulcresswell committed:
Add zenity to extra UM packages

scwhitehouse committed:
wsBcBAABCAAQBQJcrIv+CRBK7hj4Ov3rIwAAdHIIAJ2OzfTz0YViUuRDi7jJnqgm
 yJRuaiLlqvsY0wKgBiYdURbn+ugx0crE7Nfr5XWC4NxOH9buqI/Szhmy2XU5bmbl
 wC1aqUpAbzgWHiFe3f3QHflrObbzZsc56Qf9kQfIu/UjZUbrLQ50rebZ7XFWhsIW
 G00rr9+We3iRq2eqJnY/wVNIgfgWZDWMPzzEvgRSlz4KlWtxp7RykFjyoSHCVlRx
 kARn51liYAWxp5VFK5n7zuWJPhYkQhFaUIc59A8wFYUbfIlfjCFAd/XfTZLaJ5Jq
 FlXV/a/k6OEHBU3Z8/bFtYPUw7ZDzOJ3UnC4tPdlmqQgeZDJfjPmA3Ny0K//Sf8=
 =A2UQ
 -----END PGP SIGNATURE-----
 

Merge pull request #17 from paulcresswell/um_with_zenity

Add zenity to extra UM packages

dpmatthews committed:
install-rose-cylc-fcm:
- handle cylc -> cylc-flow repository name change
- restrict cylc releases to cylc 7.x

dpmatthews committed:
Ubuntu: Remove python-gi since it breaks rosie go

dpmatthews committed:
Don't install Rose 2 releases

dpmatthews committed:
Update FCM keywords

dpmatthews committed:
install-rose-meta: update location for UM metadata

Commit:
Add new keyword for UM meta project

scwhitehouse committed:
wsBcBAABCAAQBQJeUAEYCRBK7hj4Ov3rIwAAdHIIAGO12Dd5VPlGwUS5cVGPEGg2
 GQRCHHK4dg6IA7/qImjLZ/4b+tx3pyamUBwCjr4d5CQK2JEpxOEdohY4kJW+RkfL
 LvEjAUSIZPb4sNb4nc2//IYg9kJNHLOSRIzSvQHvF1g4mt0snFV5jlocJZTSHNJd
 rT1BbEeKwgT8G+/OxNtKvfufy7Fv8KwvvWTbCxD0S835s3Ro3eqq4N2uXTTXuWzq
 0JKZZ/XJGnyMwcv+GK+Y4SJMmt11r3vOgzQp6xXvXPA8qmm0EVnUYScuANWRIe99
 OTnxQ9D1E3fxcb7YuiYVUNwYAQCR50DM1Ao23UGoa94jVnu8cw2wrTkZmvaAeTY=
 =2bfl
 -----END PGP SIGNATURE-----
 

Merge pull request #19 from stevewardle/fix_keyword

Add new keyword for UM meta project

dpmatthews committed:
Update FCM keywords

dpmatthews committed:
Add installation error count
Set graphics controller to preferred "vmsvga" VirtualBox setting
Use Subversion 1.10 where possible
Replace rose bush with cylc review
Remove Ubuntu 1404 & Centos 6

dpmatthews committed:
u1804: docs build no longer working so disable for the moment

theabro committed:
Merge remote-tracking branch 'upstream/master'

theabro committed:
initial add of ansible files - needs more work

theabro committed:
about to try running

theabro committed:
moved some files around

theabro committed:
seems to be a working system now

theabro committed:
ignore inventory.ini file

theabro committed:
replaced functionality

theabro committed:
further tidy up

theabro committed:
pulled latest and resolved conflicts

theabro committed:
initial commit of AWS Vagrantfile

theabro committed:
fixes from merge

theabro committed:
added symlink file

theabro committed:
added documentation

theabro committed:
Merge pull request #1 from theabro/metomi-playbook

Metomi playbook

theabro committed:
tidy up

theabro committed:
updated .gitignore

theabro committed:
remove ansible file

theabro committed:
clean up

theabro committed:
further tweaks following additional aws tests

theabro committed:
further tweaks

theabro committed:
UM installation tested and working on t2.2xlarge

theabro committed:
further tweaks to README

theabro committed:
updated README

theabro committed:
updated README

theabro committed:
updated README

theabro committed:
further tweaks

theabro committed:
tidy up

theabro committed:
reset master Vagrantfile

theabro committed:
fix minor change

theabro committed:
minor correction to highlight may not have a file ext.

theabro committed:
minor correction to highlight may not have a file ext.

theabro commented:
Final tweaks made for AWS - now ready for review.

theabro committed:
further comment text

theabro committed:
tweak to ordering of text

theabro committed:
typo fix

dpmatthews commented on Vagrantfile.aws_ubuntu-1804:
```diff
@@ -0,0 +1,64 @@
+Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
+
+  # when provisioning symlink /home/ubuntu to /home/vagrant
+  config.vm.provision :shell, path: "vagrant_symlink.sh"
```
I think this would work
```
  config.vm.provision :shell, inline: "ln -s /home/ubuntu /home/vagrant"
```
Any reason not to change it to this?
It would allow `vagrant_symlink.sh` to be removed.
According to the docs it runs privileged by default: https://www.vagrantup.com/docs/provisioning/shell#privileged

dpmatthews reviewed (changes requested):
Just one suggestion.
Otherwise looks OK (but untested by me).

theabro committed:
fixes for install and add X2Go as connection option

theabro committed:
fix locale error, tweak to ensure 30GB volume size by default, give box a name

theabro committed:
further text for README

theabro commented:
Thanks Dave - I've made that change and it works fine. 

In doing some further tests I realised that the use of `sudo -u vagrant` causes issues on AWS as the username is `ubuntu`. I have fixed this by using `sudo -u $(logname)` instead. I have tested this on VirtualBox instances of Ubuntu 18.04, Centos 7, & Fedora 28, and this has not caused any additional problems (although both Centos and Fedora had different issues which occurred identically either using the metomi or my versions.

I've also solved my issues with the setting of the locale. I have inserted additional commands in the error trap for the `update-locale LANG=en_GB.utf8` line. This now means that the VM provisions on AWS without errors.

I've also tweaked the Vagrantfile for AWS to use a 30GB disk by default (this is still in the free tier) as it allows for moving the instance to a better instance without needing to change the volume size (8GB is not enough for most uses). I've also given the VM the name "metomi-vms" that will appear in the EC2 instance list on the AWS Console.

Finally I've added a script to include the X2Go desktop connection. This is another optional command to the `install.sh` script to install the required packages. This then means that a graphical connection is allowed, and text has been added to the README to explain how to use this option.

theabro commented:
Also, I've renamed the python terminal `pyterm` rather than `conda` in `install-iris`, as the current settings cause issues with using the `conda` command later.

theabro committed:
tweaks to README

theabro committed:
tweaks to README

theabro committed:
need to include mo_pack in Iris install due to packing used in some files

theabro commented:
I've also added `mo_pack` to the Iris install script as this is needed to open some packed files as are sometimes output from the UM

dpmatthews commented on README.md:
```diff
@@ -117,3 +118,132 @@ When you resize the VirtualBox window (e.g. to mazimise it) the display resoluti
+
+There are many different sizes of VM to choose from (known as [instance types](https://aws.amazon.com/ec2/instance-types/)), some of which will be eligible for the free tier, e.g. `t2.micro` that has 1 CPU and 1GB of memory. To be able to run the UM you will need to select a larger type, such as `t2.medium`(2 CPUs and 4GB of memory) or `t2.large`(2 CPUs and 8GB of memory). This is changed in the  ...
+
+The hard disk size of the VM can be set to 30GB in the Vagrantfile in the `aws.block_device_mapping` setting. This can be changed as required. The default `t2.micro` size is 8GB.
```
30GB is now the default?

theabro replied:
Yes, 30GB is the default. This is consistent with the current VirtualBox/Vagrant VM settings and is still part of the AWS free tier so wouldn't charge. However it does mean that no subsequent volume resizing is necessary via the AWS console.

dpmatthews replied:
I just meant that I thought it would be clearer to change "can be set" to "is set"

theabro replied:
I've tweaked the text which hopefully makes this clearer.

dpmatthews commented on README.md:
````diff
@@ -117,3 +118,132 @@ When you resize the VirtualBox window (e.g. to mazimise it) the display resoluti
+```
+source aws-credentials
+```
+Once you have created (& potentially added) the security group to your AWS Vagrantfile, you should edit the top-level Vagrantfile to point to `Vagrantfile.aws_ubuntu-1804`. Then provision the VM by
````
Presumably this would work as well?
```
export VAGRANT_VAGRANTFILE=Vagrantfile.aws_ubuntu-1804
```
If you added this to `aws-credentials` this would avoid the need to edit `Vagrantfile`

theabro replied:
Thanks for the suggestion - yes, that works as well and makes things cleaner as it doesn't involve editing the Vagrantfile. I've incorporated it in my last change.

theabro replied:
I've tweaked the text which hopefully makes this clearer.

theabro committed:
included suggestion for specifying Vagrantfile to use

theabro commented:
Thanks for the suggestion - yes, that works as well and makes things cleaner as it doesn't involve editing the Vagrantfile. I've incorporated it in my last change.

theabro committed:
clarified disk size

theabro committed:
changed tense

dpmatthews commented on README.md:
```diff
@@ -117,3 +118,133 @@ When you resize the VirtualBox window (e.g. to mazimise it) the display resoluti
+
+## Amazon AWS
+
+It is possible to run using Vagrant on an Amazon AWS EC2 virtual machine. To do this you will need to install the the [`vagrant-aws`](https://github.com/mitchellh/vagrant-aws) plugin. You do not need VirtualBox. You should ensure that you are using a recent version of Vagrant to enable the AWS plugin to work, and you will first need to run the command
```
Typo: "the the"

theabro replied:
now fixed

dpmatthews commented on README.md:
```diff
@@ -117,3 +118,133 @@ When you resize the VirtualBox window (e.g. to mazimise it) the display resoluti
+
+### Chose your region
+
+On the [AWS console](https://aws.amazon.com/) you should change your region to the one where you want the VM to be provisioned by using the drop-down menu on the top right of the page. The defualt settings may put you in `us-east-2` (US East (Ohio)), but you may want to change this to, e.g., London (or `eu-west-2`). 
```
Typo: "defualt"

theabro replied:
now fixed

theabro committed:
fixed typos

dpmatthews reviewed (approved):
Looks good.
I haven't tested AWS but I've checked the changes still work with Virtualbox.
